frontend/src/types.ts

```diff
@@ -737,22 +739,33 @@ export interface CohortPropertyFilter extends BasePropertyFilter {
     value: number
 }
 
+/** Sync with plugin-server/src/types.ts */
 export interface GroupPropertyFilter extends BasePropertyFilter {
     type: PropertyFilterType.Group
     group_type_index?: number | null
     operator: PropertyOperator
 }
 
+/** Sync with plugin-server/src/types.ts */
 export interface FeaturePropertyFilter extends BasePropertyFilter {
     type: PropertyFilterType.Feature
     operator: PropertyOperator
 }
 
+/** Sync with plugin-server/src/types.ts */
 export interface HogQLPropertyFilter extends BasePropertyFilter {
     type: PropertyFilterType.HogQL
     key: string
 }
 
+/** Sync with plugin-server/src/types.ts */
+export interface EventsFilter extends BasePropertyFilter {
+    type: PropertyFilterType.Events
+    properties: EventPropertyFilter[]
+    operator: PropertyOperator.Occurred
+}
```

Comment:
I feel pretty strongly that `Occurred` should not be a `PropertyOperator`. All the rest of `PropertyOperator`s operate on properties… this does not. I think something like this might better fit into the existing structure:
```suggestion
export interface EventsFilter extends BasePropertyFilter {
    type: PropertyFilterType.EventOccurred
    properties: EventPropertyFilter[]
}
```
BTW, do you have plans for expanding this beyond "occurred or not"?